client/cs_benchmark.cpp

```diff
@@ -260,20 +262,21 @@ void CLIENT_STATE::start_cpu_benchmarks() {
     remove_benchmark_file(BM_TYPE_INT);
     cpu_benchmarks_start = dtime();
 
-    if (benchmark_descs) {
-        free(benchmark_descs);
+    if (!benchmark_descs.empty()) {
+        benchmark_descs.clear();
     }
     bm_ncpus = ncpus;
-    benchmark_descs = (BENCHMARK_DESC*)calloc(bm_ncpus, sizeof(BENCHMARK_DESC));
     benchmarks_running = true;
 
     for (i=0; i<bm_ncpus; i++) {
+        benchmark_descs.push_back(BENCHMARK_DESC());
+
         benchmark_descs[i].ordinal = i;
         benchmark_descs[i].done = false;
         benchmark_descs[i].error = false;
 #ifdef _WIN32
         benchmark_descs[i].handle = CreateThread(
-            NULL, 0, win_cpu_benchmarks, benchmark_descs+i, 0,
+            NULL, 0, win_cpu_benchmarks, &benchmark_descs, 0,
```

Comment:
Shouldn't that be `&benchmark_descs[i]`?

Reply:
Good catch! Thank you again.